### DISCUSSION TITLE
Discuss inflection use cases to better scope the project

### DISCUSSION THREAD

**Discussion by nciric** [unicode-org/inflection#26] [Ideas]
Hi all,
as we discussed in our meeting on 2024-03-19, we should list all use cases to better scope the project.

Please use simple syntax to improve clarity.

A couple of examples to start, with the inflection feature needed to implement them:

1. Noun cases:
```
a. Your flight from {from_location} to {to_location} is delayed.
b. Hi {first_name}, your order from {restaurant} is ready for pickup."
```

For case 1.a, both from_location and to_location have to be in a proper case.

For case 1.b, first_name should be in vocative in Serbian. The current localization solution is to either drop the name, or change the message to something that avoids inflection, e.g. "Order from {restaurant} is ready for pickup for the user named {first_name}" which feels wrong to a native speaker.

**Comment by richgillam** [comment]
```
Welcome to {app name}
```
The inflection here is implicit-- The sentence addresses the user, and "Welcome" may need to inflect based on the user's gender (which you'd have to get from some external source).

The "your" in both of Nebojsa's examples above might inflect in the same way in any language where the second person singular pronoun has gender.

```
{Person} has sent you {their} high scores.
{Person} is on {their} way with your order.
```

The pronouns in both of the above sentences may need to inflect based on the gender of `{person}`, which you'd have to get from some external source.

```
The {device} is {condition}.
```

That is, "The light is on" or "The door is unlocked".  The condition may have to inflect based on the device.  "The" and "{device}" may also need to change when juxtaposed.

```
You have ordered {quantity} {size} {item}.
```
That is, "You have ordered 3 large pizzas" or "You have ordered 1 small salad".  The quantity may affect the grammatical number of both the size and the item; the item may affect the gender of the size (and maybe of the quantity).

**Reply by BrunoCartoni** [reply]
Very similar (but adding extra-examples):

`{you} have been {disconnected}`

Both "you" and "disconnected" should agree with the gender of the user.

**Reply by nciric** [reply]
Also, gender related and we see it when welcoming the user to the device.

`Welcome {username}`

The {username} is optional, welcome is enough to require gender in Serbian or Romanian.

@BrunoCartoni what kind of grammatical type is "welcome" in this sentence?

**Reply by nciric** [reply]
@BrunoCartoni I am trying to come up with an example language where gender for {you} is needed. The particip/verb part I agree.

**Comment by richgillam** [comment]
I think there are a lot of interesting cases involving date, times, date/time intervals, and durations:

```
Today is {date}
...on {date}
...before/after {date}
from {date} to {date}
n {intervals} before/after {date}
today/tomorrow/{day name}/{full date}
next {day/month name}
last {day/month name}
in the next {duration}
for the next {duration}
in the previous {duration}
n {intervals} from now
n {intervals} ago
every {interval}/every other {interval}/every {third, fourth, etc.} {interval}
{once/twice/n times} per {interval}
```

**Comment by BrunoCartoni** [comment]
![an_a_example](https://github.com/unicode-org/inflection/assets/161062932/c2899061-7754-4f1b-84c2-6afb281b5ae8)

**Comment by BrunoCartoni** [comment]
Arabic for examples: https://screenshot.googleplex.com/53kJwgj5QqtSi9a


On Tue, Apr 2, 2024 at 5:22 PM Nebojša Ćirić ***@***.***>
wrote:

> @BrunoCartoni <https://github.com/BrunoCartoni> I am trying to come up
> with an example language where gender for {you} is needed. The
> particip/verb part I agree.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/inflection/discussions/26#discussioncomment-8985685>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BGM2AFC3PD74ALKU2MICQBLY3LEM3AVCNFSM6AAAAABE6DRDFSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DSOBVGY4DK>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Bruno Cartoni |  (he/him) | Staff Linguist | Pride at Google Zürich Lead |
 ***@***.*** |  +41.79.246.80.46

**Comment by nciric** [comment]
An example where UI looks weird due to inability to inflect the name of the product (Tasks in this case): 

![image](https://github.com/unicode-org/inflection/assets/4088362/fcbd0939-21b6-4a08-b2a4-63365e6c9eda)

This use case could be filed under:
1. Product name (noun) inflection
2.  Internal override for the end client

**Comment by nciric** [comment]
Added a pull request to document decisions from our meeting #27
